### PR TITLE
Add a few unit tests for MspHelper class

### PR DESCRIPTION
**Pull request by Benky**
## Description

This PR adds a few unit tests for `MspHelper` class (more to come). Most of the data in unit tests are randomly generated, so we can test for various length / number of cases.

I've also added a few commits that simplify `encode_message_v1` and handling of `MSP_BOARD_INFO` (for better readability and maintainability). 

## Notes
I can separate commits with simplification of the methods to separate PR is required.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — APPROVED** [review]

**Review by nerdCopter — APPROVED** [review]

**Comment by haslinghuis** [comment]
@Benky is this is false positive? Notices while working on another PR and see it's in the above test also.

```
stderr | test/js/msp/MSPHelper.test.js > MspHelper > process_data > refuses to process data with crc-error
code: 4 (MSP_BOARD_INFO) - crc failed
```

**Comment by Benky** [comment]
Hey @haslinghuis - first of all, sorry for late response.

It's actually an expected thing - it's generated by https://github.com/betaflight/betaflight-configurator/blob/master/src/js/msp/MSPHelper.js#L1645 which gets triggered during this test : https://github.com/betaflight/betaflight-configurator/blob/165735159874441949d6c417dc0226fb1e3f0f9d/test/js/msp/MSPHelper.test.js#L14 which actually tests handling of messages with failed CRC checksums.

**Pull request merged**
